event_feed.go

```diff
@@ -0,0 +1,78 @@
+package fauna
+
+import (
+	"encoding/json"
+)
+
+// EventFeed represents an event feed subscription.
+type EventFeed struct {
+	client *Client
+
+	source EventSource
+
+	decoder *json.Decoder
+
+	opts       []FeedOptFn
+	lastCursor string
+}
+
+func newEventFeed(client *Client, source EventSource, opts ...FeedOptFn) (*EventFeed, error) {
+	feed := &EventFeed{
```

Comment:
we should validate only one of the cursor or start ts was provided here, but not both